toplevel/opttoploop.ml

```diff
@@ -165,7 +165,8 @@ let load_lambda ppf (size, lam) =
     else Filename.temp_file ("caml" ^ !phrase_name) ext_dll
   in
   let fn = Filename.chop_extension dll in
-  Asmgen.compile_implementation ~toplevel:need_symbol fn ppf (size, slam);
+  Asmgen.compile_implementation ~sourcefile:"" ~toplevel:need_symbol fn ppf
```

Comment:
Thinking more about it, this would justify changing the `compile_implentation` from `sourcefile` to `source_provenance`. @chambart, is there a good reason for passing `~sourcefile:"pack"` [in Asmpackager](https://github.com/trefis/ocaml/blob/ocamlnat/asmcomp/asmpackager.ml#L94)?